File: scanner/corpora/python/positive/chacha20.py
"""ChaCha20-Poly1305 encryption using the cryptography library."""
from cryptography.hazmat.primitives.ciphers.aead import ChaCha20Poly1305

key = bytes(range(32))  # 256-bit key
nonce = bytes(range(12))  # 96-bit nonce
cipher = ChaCha20Poly1305(key)
plaintext = b"Secret message for ChaCha20"
aad = b"associated authenticated data"
ciphertext = cipher.encrypt(nonce, plaintext, aad)
decrypted = cipher.decrypt(nonce, ciphertext, aad)
print(decrypted)

File: scanner/corpora/go/positive/chacha20.go
package main

import (
	"crypto/rand"
	"golang.org/x/crypto/chacha20poly1305"
	"io"
	"fmt"
)

func chacha20Encrypt(plaintext, key []byte) ([]byte, error) {
	aead, err := chacha20poly1305.NewX(key)
	if err != nil {
		return nil, err
	}
	nonce := make([]byte, aead.NonceSize())
	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
		return nil, err
	}
	return aead.Seal(nonce, nonce, plaintext, nil), nil
}

func main() {
	key := make([]byte, chacha20poly1305.KeySize)
	rand.Read(key)
	data := []byte("Secret ChaCha20 message")
	ct, err := chacha20Encrypt(data, key)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Ciphertext length: %d\n", len(ct))
}
